p2p/utils.py

```diff
@@ -30,3 +34,16 @@ def get_devp2p_cmd_id(msg: bytes) -> int:
     as an integer.
     """
     return rlp.decode(msg[:1], sedes=rlp.sedes.big_endian_int)
+
+
+class RunningServices:
```

Comment:
Maybe this should be a subclass of [`collections.UserList`](https://docs.python.org/3/library/collections.html#collections.UserList) so that you can use it on the `Node` as the actual `_auxiliary_services` property as well as locking down the list items while the context manager is active.